repositories/RoutingRepository.js

```diff
@@ -0,0 +1,171 @@
+const { head, invert, map } = require("lodash/fp");
+const mapping = {
+  QuestionPageId: "questionPageId",
+  ElseDestination: "elseDestination",
+  RoutingRuleSetId: "routingRuleSetId",
+  RuleDestination: "ruleDestination",
+  RoutingRuleId: "routingRuleId",
+  AnswerId: "answerId",
+  OptionId: "optionId",
+  ConditionId: "conditionId"
+};
+const mapFields = require("../utils/mapFields");
+const fromDb = mapFields(mapping);
+const toDb = mapFields(invert(mapping));
+
+const db = require("../db");
+
+const {
+  updateConditions,
+  createOrRemoveValue
+} = require("./strategies/routingStrategy");
+
+const Routing = require("../db/Routing");
+
+module.exports.findRoutingRuleSetByQuestionPageId = function findRoutingRuleSetByQuestionPageId(
+  where = {}
+) {
+  return Routing.findAllRoutingRuleSets()
+    .where({ isDeleted: false })
+    .where(where)
+    .first()
+    .then(fromDb);
+};
+
+module.exports.findAllRoutingRules = function findAllRoutingRules(where = {}) {
+  return Routing.findAllRoutingRules()
+    .where({ isDeleted: false })
+    .where(where)
+    .then(map(fromDb));
+};
+
+module.exports.findAllRoutingConditions = function findAllRoutingConditions(
+  where = {}
+) {
+  return Routing.findAllRoutingConditions()
+    .where(where)
+    .then(map(fromDb));
+};
+
+module.exports.findAllRoutingConditionValues = function findAllRoutingConditionValues(
+  where = {}
+) {
+  return Routing.findAllRoutingConditionValues()
+    .where(where)
+    .map(result => result.OptionId);
+};
+
+module.exports.insertRoutingRuleSet = function insertRoutingRuleSet({
```

Comment:
Can do but I'm not sure it would make the implementation much cleaner as it would just change the order of the names 

i.e. Routing.findAllRoutingRules() => RoutingRules.findAll()

and it would also just spread the functions over a number of files. If we're sure that implementation is better I can refactor though.